src/fetch/fetchWithRetries.js

```diff
@@ -86,7 +86,12 @@ function fetchWithRetries(
             retryRequest();
           } else {
             // Request was not successful, giving up.
-            reject(response);
+            var error: any = new Error(sprintf(
+              'fetchWithRetries(): Gave up after %s re-tries.',
```

Comment:
Maybe we can tune the error message a bit. This is the exact same error message one gets after the last attempted timed-out. In this case we tried X times and the final attempted returned an error status code.